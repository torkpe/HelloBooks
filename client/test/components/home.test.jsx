import React from 'react';
import toJson from 'enzyme-to-json';
import { configure, shallow, mount } from 'enzyme';
import Adapter from 'enzyme-adapter-react-15';
import render from 'react-test-renderer';
import { Provider } from 'react-redux';
import configureMockStore from 'redux-mock-store';

import connectedHome, { Home } from '../../components/Home.jsx';
import storeConfiguration from '../../store/store';

configure({ adapter: new Adapter() });

describe('<Home />', () => {
  const props = {
    auth: {
      user: {
        star: 'gh',
        id: 1
      },
    },
    getBooks: jest.fn(),
    books: [{
      title: 'eze goes to school',
      description: 'a boy from the village who finally went to school',
      id: 1,
      cover: 'xcgvhjkhcvbkjhgcvbjhcv',
      pdf: 'cghjklhgcfhgjkhgjh',
      quantity: 1,
      genre: 'eductaional'
    }],
    fetching: false,
    userId: 1,
    borrowedBooks: [{}]
  };
  const initialState = {
    fetching: false,
    books: [],
    errors: '',
  };
  const mockStore = configureMockStore();
  let mountedComponent;
  let store;
  let wrapper;
  beforeEach(() => {
    store = mockStore(storeConfiguration());
    mountedComponent = mount(<Home {...props} />);
    wrapper = mount(<Provider store={storeConfiguration()}><connectedHome /></Provider>);
  });
  it('renders <Home /> component', () => {
    const tree = toJson(mountedComponent);
    expect(tree).toMatchSnapshot();
  });
  it('should be defined', () => {
    expect(mountedComponent).toBeDefined();
  });
  it('should check Prop matches with initialState', () => {
    const tree = toJson(mountedComponent);
    expect(Object.keys(tree.node.props).length).toBeGreaterThanOrEqual(1);
  });
  it('checks if loading is true', () => {
    props.fetching = true;
    const shallowComponent = shallow(<Home {...props} />);
    expect(shallowComponent.instance().props.fetching).toEqual(true);
  });
  it('checks if books length is less than 1', () => {
    props.books = {};
    const shallowComponent = shallow(<Home {...props} />);
    expect(Object.keys(shallowComponent.instance().props.books).length < 1).toEqual(true);
  });
});