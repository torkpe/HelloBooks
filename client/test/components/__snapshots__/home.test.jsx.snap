// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Home /> renders <Home /> component 1`] = `
<Home
  auth={
    Object {
      "user": Object {
        "id": 1,
        "star": "gh",
      },
    }
  }
  books={
    Array [
      Object {
        "cover": "xcgvhjkhcvbkjhgcvbjhcv",
        "description": "a boy from the village who finally went to school",
        "genre": "eductaional",
        "id": 1,
        "pdf": "cghjklhgcfhgjkhgjh",
        "quantity": 1,
        "title": "eze goes to school",
      },
    ]
  }
  borrowedBooks={
    Array [
      Object {},
    ]
  }
  fetching={false}
  getBooks={
    [MockFunction] {
      "calls": Array [
        Array [],
      ],
    }
  }
  userId={1}
>
  <div
    className="mdl-grid "
  >
    <div
      className="mdl-cell mdl-cell--1-col"
    />
    <div
      className="mdl-cell mdl-cell--10-col"
    >
      <div
        className="mdl-grid "
      >
        <div
          className="mdl-cell mdl-cell--2-col"
        />
      </div>
      <div
        className="contents"
      >
        <span
          className="star"
        >
          Membership level: 
          gh
        </span>
        <h5>
          All Books
        </h5>
        <hr />
      </div>
      <div
        className="ask"
      />
      <div
        className="mdl-grid"
      >
        <Books
          auth={
            Object {
              "user": Object {
                "id": 1,
                "star": "gh",
              },
            }
          }
          book={
            Object {
              "cover": "xcgvhjkhcvbkjhgcvbjhcv",
              "description": "a boy from the village who finally went to school",
              "genre": "eductaional",
              "id": 1,
              "pdf": "cghjklhgcfhgjkhgjh",
              "quantity": 1,
              "title": "eze goes to school",
            }
          }
          books={
            Array [
              Object {
                "cover": "xcgvhjkhcvbkjhgcvbjhcv",
                "description": "a boy from the village who finally went to school",
                "genre": "eductaional",
                "id": 1,
                "pdf": "cghjklhgcfhgjkhgjh",
                "quantity": 1,
                "title": "eze goes to school",
              },
            ]
          }
          borrowedBooks={
            Array [
              Object {},
            ]
          }
          fetching={false}
          getBooks={
            [MockFunction] {
              "calls": Array [
                Array [],
              ],
            }
          }
          key="1"
          userId={1}
        >
          <div
            className="mdl-cell mdl-cell--4-col"
          >
            <div
              className="demo-card-square mdl-card mdl-shadow--2dp home-card contents"
            >
              <div
                className="mdl-card__title mdl-card--expand"
              >
                <h2
                  className="mdl-card__title-text"
                >
                  eze goes to school
                </h2>
              </div>
              <img
                alt="Book cover"
                src="xcgvhjkhcvbkjhgcvbjhcv"
              />
              <div
                className="mdl-card__supporting-text"
              >
                a boy from the village who finally went to school
              </div>
              <div
                className="mdl-card__actions mdl-card--border"
              >
                <Link
                  className="
    mdl-button
    mdl-button--colored
    mdl-js-button
    mdl-js-ripple-effect
  "
                  onlyActiveOnIndex={false}
                  style={Object {}}
                  to="single/1"
                />
              </div>
            </div>
          </div>
        </Books>
      </div>
      <div
        className="mdl-grid "
      >
        <div
          className="mdl-cell mdl-cell--2-col"
        />
      </div>
    </div>
    <div
      className="mdl-cell mdl-cell--1-col"
    />
  </div>
</Home>
`;
